skglm/tests/test_estimators.py

```diff
@@ -97,27 +97,38 @@ def test_check_estimator(estimator_name):
 # Test if skglm solver returns the coefficients
 @pytest.mark.parametrize("estimator_name", dict_estimators_ours.keys())
 @pytest.mark.parametrize('X', [X, X_sparse])
-def test_estimator(estimator_name, X):
+@pytest.mark.parametrize('fit_intercept', [True, False])
+def test_estimator(estimator_name, X, fit_intercept):
     if estimator_name == "GeneralizedLinearEstimator":
         pytest.skip()
     estimator_sk = dict_estimators_sk[estimator_name]
     estimator_ours = dict_estimators_ours[estimator_name]
+    # TODO This seems a bit unusal, maybe to discuss
+    if fit_intercept:
+        estimator_sk.fit_intercpet = True
```

Comment:
There is a typo `intercpet`

That's why we try to avoid such designs, they can fail silently.